crypto/evp/evp_enc.c

```diff
@@ -1031,7 +1031,7 @@ int EVP_CIPHER_CTX_set_padding(EVP_CIPHER_CTX *ctx, int pad)
     params[0] = OSSL_PARAM_construct_uint(OSSL_CIPHER_PARAM_PADDING, &pd);
     ok = evp_do_ciph_ctx_setparams(ctx->cipher, ctx->algctx, params);
 
-    return ok != 0;
+    return ok > 0;
```

Comment:
Looks like there are quite a few similar instances of this bug. I'm looking at: EVP_CIPHER_CTX_original_iv, EVP_CIPHER_CTX_iv, EVP_CIPHER_CTX_iv_noconst, EVP_CIPHER_CTX_get_num, EVP_CIPHER_CTX_get_key_length, EVP_CIPHER_CTX_set_num

Reply:
With pleasure to do that. But I am confused while reviewing that where is the implementation or definition of functions like evp_do_ciph_ctx_setparams. I concluded just by the comment in the header file and usage in other places.

Reply:
They are implemented here:

https://github.com/openssl/openssl/blob/2c2724476ef50b8926b033f009bdfc85ac3f1816/crypto/evp/evp_utils.c#L20-L81

Notably all of these functions can return EVP_CTRL_RET_UNSUPPORTED (which is defined as -1).